tree/tree/src/TTreeSQL.cxx

```diff
@@ -375,108 +382,98 @@ void TTreeSQL::CreateBranch(const TString &branchName, const TString &typeName)
 ////////////////////////////////////////////////////////////////////////////////
 /// determine leaf description string
 
-TString TTreeSQL::CreateBranches(TSQLResult * rs)
+void TTreeSQL::CreateBranches()
 {
-   if(!rs) return "";
+   TList * columns = fTableInfo->GetColumns();
+   if(!columns) return;
+
+   TIter next(columns);
 
-   Int_t rows;
-   TString type;
-   TString res;
    TString branchName;
+   TString type;
    TString leafName;
-   Int_t prec=0;
    TBranch * br = 0;
-   rows = rs->GetRowCount();
-   TString decl;
-   TString prevBranch;
-
-   for( int i=0; i < rows; ++i ) {
-      TSQLRow * row = rs->Next();
-      if (!row) continue;
-      type = row->GetField(1);
-      Int_t index = type.First('(');
-      if(index>0){
-         prec = atoi(type(index+1,type.First(')')-1).Data());
-         type = type(0,index);
-      }
-      branchName = row->GetField(0);
+   TSQLColumnInfo * info;
+   while ( (info = ((TSQLColumnInfo*) next()) ))
+   {
+      type = info->GetTypeName();
+      branchName = info->GetName();
+
+
       Int_t pos;
       if ((pos=branchName.Index("__"))!=kNPOS) {
-         leafName = branchName(pos+2,branchName.Length());
-         branchName.Remove(pos);
+          leafName = branchName(pos+2,branchName.Length());
+          branchName.Remove(pos);
       } else {
-         leafName = branchName;
+          leafName = branchName;
       }
-      if (prevBranch.Length()) {
-         if (prevBranch != branchName) {
-            // new branch let's flush.
-            if (decl.Length()) decl.Remove(decl.Length()-1);
-            br = TTree::Branch(prevBranch,0,decl);
-            br->ResetAddress();
-
-            (br->GetBasketEntry())[0] = 0;
-            (br->GetBasketEntry())[1] = fEntries;
-
-            br->SetEntries(fEntries);
-
-            //++(br->fNBaskets);
-            br->GetListOfBaskets()->AddAtAndExpand(CreateBasket(br),0);
 
-            prevBranch = branchName;
-            decl = "";
-         }
-      } else {
-         prevBranch = branchName;
-      }
+      TString str;
+      int i;
+      unsigned ui;
+      double d;
+      float f;
+
+      br = 0;
+
+      if(type.CompareTo("varchar",TString::kIgnoreCase)==0 ||
+         type.CompareTo("varchar2",TString::kIgnoreCase)==0 ||
+         type.CompareTo("char",TString::kIgnoreCase)==0 ||
+         type.CompareTo("longvarchar",TString::kIgnoreCase)==0 ||
+         type.CompareTo("longvarbinary",TString::kIgnoreCase)==0 ||
+         type.CompareTo("varbinary",TString::kIgnoreCase)==0 ||
+         type.CompareTo("text",TString::kIgnoreCase )==0 ) {
+         br = TTree::Branch(leafName,&str);
```

Comment:
I am not sure, if here and in several other places `TTree::Branch(leafName,&str);` calls with local-variable pointer is correct. @pcanal may say more

Reply:
I think it's ok because the branch address is reset below. Before the code was doing something similar with a leaflist and a null address but that doesn't work for TString. I could probably use the templated version of Branch with a null address to be more clear...

Reply:
Yes, the ResetAddress makes it okay (since the address is forgotten).